beacon-chain/cache/sync_committee.go

```diff
@@ -162,6 +181,9 @@ func (s *SyncCommitteeCache) UpdatePositionsInCommittee(syncCommitteeBoundaryRoo
 
 	s.lock.Lock()
 	defer s.lock.Unlock()
+	if clearCount != s.cleared.Load() {
+		return errors.New("cache rotated during async committee update operation")
```

Comment:
Any thoughts on logging the error and returning nil vs. returning the error?
Question is whether this error should fail block processing

Reply:
It's a bummer that we even need to consider this, but I agree that I don't want to create an issue in block processing, because strict ordering of cache updates so far seems to only be an issue in tests. But I don't like how the current design requires so much care to avoid races, so I'm inclined to go ahead and log for now, then return to improving this design when we audit cache implementations.